llama_recipes/llama_finetuning.py

```diff
@@ -81,6 +81,7 @@ def main(**kwargs):
     if train_config.enable_fsdp:
         setup()
         # torchrun specific
+        import os
```

Comment:
move this up to the top-of-file imports

Reply:
this is a pretty cursed bug https://replicate.com/p/fxtkbl3btli7tclyvlcjua6eum

Traceback (most recent call last):
File "/src/llama_recipes/llama_finetuning.py", line 369, in <module>
fire.Fire(main)
File "/usr/local/lib/python3.11/site-packages/fire/core.py", line 141, in Fire
component_trace = _Fire(component, args, parsed_flag_args, context, name)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/fire/core.py", line 475, in _Fire
component, remaining_args = _CallAndUpdateTrace(
^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
component = fn(*varargs, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^
File "/src/llama_recipes/llama_finetuning.py", line 84, in main
local_rank = int(os.environ["LOCAL_RANK"])
^^
UnboundLocalError: cannot access local variable 'os' where it is not associated with a value


in that version line 3 of llama_finetuning.py is "import os", and yet. I think it's because of some multiprocessing nonsense?